addons/test_mail/tests/test_mail_message.py

```diff
@@ -291,6 +293,17 @@ def test_mail_message_access_create_reply(self):
         self.message.write({'partner_ids': [(4, self.user_employee.partner_id.id)]})
         self.env['mail.message'].sudo(self.user_employee).create({'model': 'mail.channel', 'res_id': self.group_private.id, 'body': 'Test', 'parent_id': self.message.id})
 
+    def test_mail_message_portal_user_access_parent_message(self):
+        message = self.test_record.message_post(
+            body='<p>This is First Message</p>', subject='Subject',
+            message_type='comment', subtype='mail.mt_note')
+        with patch.object(MailTestSimple, 'check_access_rights', return_value=True):
```

Comment:
I just don't like that we use a mock here, but I don't really have a better idea.

Reply:
I don't understand. It seems we need a test for people not having access to a message that post a new message based on it for references. I am not sure this test tests that behavior as patching check_access_rights with return value True will enables reading on the message.

Anyway we need to 1/ explicitly check that portal user cannot read the new message, 2/ post a message and check everything ran smoothly with message.message-id present in references.

This test is too much high level to ensure it protects against future modifications.